NICE.Identity.Management/ClientApp/src/components/FilterSuggestions/FilterSuggestions.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { Component, ElementType } from "react";
+import { Panel } from "@nice-digital/nds-panel";
+import { Input } from "@nice-digital/nds-input";
+import styles from "./FilterSuggestions.module.scss";
+
+type FilterSuggestionsProps<T> = {
+	label: string;
+	data: T[];
+	qaSelExtract: string;
+	onInputChange: (searchQuery: string) => void;
+	onResultClick: (item: T) => void;
+	elementType: ElementType;
+};
+
+type FilterSuggestionsState = {
+	activeSuggestionIndex: number;
+};
+
+export class FilterSuggestions<T> extends Component<
+	FilterSuggestionsProps<T>,
+	FilterSuggestionsState
+> {
+	constructor(props: FilterSuggestionsProps<T>) {
+		super(props);
+
+		this.state = {
+			activeSuggestionIndex: 0,
+		};
+	}
+
+	typingTimer = 0;
+	suggestionIdPrefix = "suggestion";
+
+	handleInputKeyDown = (
+		e: React.KeyboardEvent<HTMLInputElement>,
+	): void | boolean => {
+		const key = e.key;
+		const suggestions = this.props.data;
+		const suggestionsCount = suggestions.length > 3 ? 3 : suggestions.length;
+		let activeSuggestionIndex = this.state.activeSuggestionIndex;
+
+		if (!suggestionsCount) {
+			return false;
+		}
+
+		if (key === "Enter") {
+			const item = suggestions[activeSuggestionIndex - 1];
+			this.props.onResultClick(item);
+			activeSuggestionIndex = 0;
+		}
+
+		if (key === "ArrowUp" || key === "ArrowDown") {
+			e.preventDefault();
+			activeSuggestionIndex =
+				key === "ArrowUp"
+					? activeSuggestionIndex === 0
+						? suggestionsCount
+						: activeSuggestionIndex - 1
+					: activeSuggestionIndex === suggestionsCount
+					? 0
+					: activeSuggestionIndex + 1;
+		}
+
+		this.setState({ activeSuggestionIndex });
+	};
+
+	handleInputChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+		const val = e.target.value.trim();
+
+		clearTimeout(this.typingTimer);
+
+		this.typingTimer = window.setTimeout(() => {
```

Comment:
we've been favouring requestAnimationFrame over setTimeout because it's more predictable and accurate regardless of what else is happening in the browser (busy tabs etc). Maybe see if it will fit your use case?